src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,367 @@
+        <View style={styles.topBarButtonsContainer}>
+          {Platform.OS === 'ios' && (
+            <TopBarIconButton
+              onPress={() => navigateBack()}
+              icon={<BackChevronStatic />}
+              style={styles.button}
+              testID="NftsInfoCarousel/BackButton"
+            />
+          )}
+          {Platform.OS === 'android' && (
+            <Pressable
+              android_ripple={android_ripple}
+              onPress={() => navigateBack()}
+              style={styles.button}
+              testID="NftsInfoCarousel/BackButton"
+            >
+              <BackChevronStatic />
+            </Pressable>
+          )}
+        </View>
+      </View>
+    )
+  }
+
+  const MainImagePlaceholder = () => {
+    return (
+      <SkeletonPlaceholder
+        borderRadius={8}
+        backgroundColor={colors.gray2}
+        highlightColor={colors.white}
+      >
+        <View
+          style={{
+            height: scaledHeight,
+            width: variables.width,
+            zIndex: -1,
+          }}
+        />
+      </SkeletonPlaceholder>
+    )
+  }
+
+  const NftImageCarousel = () => {
+    return (
+      <View style={styles.nftImageCarouselContainer}>
+        <ScrollView
+          showsHorizontalScrollIndicator={false}
+          horizontal={true}
+          contentContainerStyle={styles.carouselScrollViewContentContainer}
+          style={styles.nftImageCarousel}
+          testID="NftsInfoCarousel/NftImageCarousel"
+        >
+          {/* Only display Nfts with metadata */}
+          {nfts
+            .filter((nft) => nft.metadata)
+            .map((nft) => {
+              return (
+                <View
+                  key={`${nft.contractAddress}-${nft.tokenId}`}
+                  style={styles.nftPreviewImageSharedContainer}
+                  testID={`NftsInfoCarousel/NftThumbnail/${nft.contractAddress}-${nft.tokenId}`}
+                >
+                  <NftThumbnail nft={nft} setActiveNft={setActiveNft} activeNft={activeNft} />
+                </View>
+              )
+            })}
+        </ScrollView>
+      </View>
+    )
+  }
+
+  // Display error screen no Nfts provided
+  if (nfts.length === 0) {
+    return <NftsLoadErrorScreen />
+  }
+
+  return (
+    <SafeAreaView testID="NftsInfoCarousel">
+      <PlatformSpecificTopBarButtons />
+      <ScrollView>
+        {activeNft.metadata && (
+          <FastImage
+            testID={`NftsInfoCarousel/NftImage-${activeNft.contractAddress}-${activeNft.tokenId}`}
+            style={[
+              {
+                height: scaledHeight,
+                width: variables.width,
+              },
+              isLoading && { borderRadius: 8 },
+            ]}
+            source={{
+              uri: activeNft.media.find((media) => media.raw === activeNft.metadata?.image)
+                ?.gateway,
+            }}
+            onLoad={(e) => {
+              setScaledHeight(
```

Comment:
what you are trying to do is layout the image so that you can determine the aspect ratio, then resize it to fit the screen right? this seems a bit complex to have a state variable and layout function...

could you do that with styles? like set the width as `variables.width` and use [resizeMode](https://reactnative.dev/docs/image#resizemode)?

Reply:
There's no support for height="auto" in RN, so I had to use this instead.